app/src/main/java/xyz/jhughes/laundry/adapters/MachineAdapter.java

```diff
@@ -135,24 +145,134 @@ public void onClick(DialogInterface dialog, int which) {
     }
 
     public void registerNotification(Machine m) {
+        //For available (green) machines
+        if (m.getStatus().equals("Available")) {
+            waitForMachine(m);
+        } else {
+            try {
+                //For machines that are already running
+                int minutesInFuture = Integer.parseInt(m.getTime().substring(0, m.getTime().indexOf(' ')));
+                int millisInFuture = minutesInFuture * 60000; //60 seconds * 1000 milliseconds
```

Comment:
This method was the one previously in the app where it checks if the washer/dryer contains a time, uses that to determine if it is running, and then creates a popup dialog asking the user if they want to start a timer.  I copied parts of it into the method below where since the dialog has already been created and the timer should start automatically, it starts the timer directly rather then calling the "notificationWithDialog" method.